lib/SIL/TypeLowering.cpp

```diff
@@ -824,6 +863,67 @@ namespace {
     }
   };
 
+  using DifferentiableSILFunctionTypeIndex =
+      std::pair<AutoDiffFunctionExtractee, unsigned>;
+  class DifferentiableSILFuncionTypeLowering final
+      : public LoadableAggTypeLowering<DifferentiableSILFuncionTypeLowering,
+                                       DifferentiableSILFunctionTypeIndex> {
+  public:
+    DifferentiableSILFuncionTypeLowering(CanType type)
+        : LoadableAggTypeLowering(type) {}
+
+    SILValue emitRValueProject(SILBuilder &B, SILLocation loc,
+                               SILValue tupleValue,
+                               DifferentiableSILFunctionTypeIndex index,
+                               const TypeLowering &eltLowering) const {
+      return B.createAutoDiffFunctionExtract(loc, index.first, index.second,
+                                             tupleValue);
+    }
+
+    SILValue rebuildAggregate(SILBuilder &B, SILLocation loc,
+                              ArrayRef<SILValue> values) const override {
+      auto fnTy = getLoweredType().castTo<SILFunctionType>();
+      auto paramIndices = fnTy->getDifferentiationParameterIndices();
+      // TODO: Retrieve the differentiation order when that is properly stored
+      // in the function type.
+      unsigned maxOrder = 1;
+      return B.createAutoDiffFunction(loc, paramIndices, maxOrder,
+                                      values.front(), values.drop_front());
+    }
+
+    void lowerChildren(SILModule &M,
```

Comment:
@marcrasi I implemented this because this is required for compiling stdlib with those new differential operators.